rfcs/text/0000-create-file-elf.md

```diff
@@ -0,0 +1,306 @@
+# 0000: Create the ELF sub-field of the File fieldset
+
+- Stage: **0 (strawperson)**
+- Date: **TBD**
+
+Create the Executable Linkable Format (ELF) sub-field, of the `file` top-level fieldset. This document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+**Stage 0**
+
+This RFC is to create the ELF sub-field within the `file.` fieldset. This will include 25 sub-fields.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| file.elf.creation_date | date | Extracted when possible from the file's metadata. Indicates when it was built or compiled. It can also be faked by malware creators. |
+| file.elf.exports.name | keyword | Name of exported symbol |
+| file.elf.exports.type | keyword | Type of exported symbol |
+| file.elf.segments | keyword | ELF object segment list. |
```

Comment:
Is this intended to be just a pure list of the segment names (i.e. `PT_LOAD`, `PT_DYNAMIC`, etc?)

Reply:
This is how the data I'm seeing is presented
```
{
  "type": "PHDR",
  "sections": []
},
{
  "type": "INTERP",
  "sections": [
    ".interp"
  ]
},
{
  "type": "LOAD",
  "sections": [
    ".interp",
    ".note.android.ident",
    ".note.gnu.build-id",
    ".dynsym",
    ".dynstr",
    ".gnu.hash",
    ".hash",
    ".gnu.version",
    ".gnu.version_r",
    ".rel.dyn",
    ".rel.plt",
    ".plt",
    ".text",
    ".ARM.exidx",
    ".ARM.extab",
    ".rodata"
  ]
},
{
  "type": "LOAD",
  "sections": [
    ".preinit_array",
    ".init_array",
    ".fini_array",
    ".data.rel.ro",
    ".dynamic",
    ".got",
    ".data",
    ".bss"
  ]
},
{
  "type": "DYNAMIC",
  "sections": [
    ".dynamic"
  ]
},
{
  "type": "NOTE",
  "sections": [
    ".note.android.ident",
    ".note.gnu.build-id"
  ]
},
{
  "type": "GNU_STACK",
  "sections": []
},
{
  "type": "ARM_EXIDX",
  "sections": [
    ".ARM.exidx"
  ]
},
{
  "type": "GNU_RELRO",
  "sections": [
    ".preinit_array",
    ".init_array",
    ".fini_array",
    ".data.rel.ro",
    ".dynamic",
    ".got"
  ]
}
```

Reply:
ok, so that tells me we have two options -- either do another `nested` field of `nested` `segments` with nested `sections` 😞 or potentially just use this as capturing the values in `type` as in your example. TBH I'm not sure either is really ideal, but the double `nested` models more closely what the relationship is for segments/sections and the latter drops that information, which could be useful.

Reply:
I updated to this
```
...
    - name: segments
      description: >
        ELF object segment list.
      type: nested
      level: extended
      fields:
        - name: type
          description: ELF object segment type.
          type: keyword
          level: extended

        - name: sections
          description: ELF object segment sections.
          type: keyword
          level: extended
```